fe/fe-core/src/main/java/com/starrocks/catalog/Column.java

```diff
@@ -761,7 +761,7 @@ public String toSqlWithoutAggregateTypeName(Map<ColumnId, Column> idToColumn) {
         if (isGeneratedColumn()) {
             String generatedColumnSql;
             if (idToColumn != null) {
-                generatedColumnSql = AstToSQLBuilder.toSQL(generatedColumnExpr.convertToColumnNameExpr(idToColumn));
+                generatedColumnSql = generatedColumnExpr.convertToColumnNameExpr(idToColumn).toSql();
```

Comment:
What is the difference between these two lines? We should not rely on the hidden logical difference between toSql and AstToSql. Can we encapsulate the difference in AstToSQLBuilder, or analyze expression in advance?

Reply:
For many type of expression, `AstToSQLBuilder .toSQL()` is the same as `Expr.toSql()` exception for `slotRef`, `MapExpr` and `ArraryExpr`. Beside the expression, `AstToSQLBuilder .toSQL()` will also be used for some `Relation` serialization mainly using for `VIEW` or `MV` context. But for generated column, we always expect that we need the unanalyzed one for the serialization SQL.

Reply:
In other word, `AstToSQLBuilder.toSQL()` is used for QueryStatement not only for Expression, so it is reasonable that we need analylzed version for it, but it is not the case for generated column expression only.